architecture-docs/adr-001.md

```diff
@@ -50,43 +51,60 @@ to do a more complete verification (currently there are a few hacks and workarou
 ## Decision
 This will be a bit big change, so it can be done in several steps:
 
-1. separate out the storage functionality from chain-abci into chain-storage crate
-
-2. TBD: separate storage process? moving enclave app wrappers to chain-abci?
-
-3. change enclave-protocol for chain-abci/storage to add sealed transaction data, and TVE to return Ok(sealed transaction data) back to chain-abci/storage
-
-4. give TQE some way to talk to chain-abci/storage (TBD: another 0MQ socket or abci_query?)
-
-5. change the first flow of TQE to use that instead of talking to TVE
-
-6. TBD: change the second flow to go through chain-abci (it'd still need to forward to TVE)
-
-7. remove storage from TVE (as it should be all handled by chain-abci/storage)
-
+* separate out the storage functionality from chain-abci into chain-storage crate
```

Comment:
What functionality should be moved into the `chain-storage` crate? Or what kind of APIs does it provide?

Reply:
ADR isn't a spec, so the concrete API will evolve -- by "storage functionality", it's mainly things that can be extracted out from the storage module of chain-abci (into more high-level things like -- getTx, checkOutputSpent, updates...).
The purposes are two: 1) decouple plain storage operations from state machine logic, 2) check the storage-invariants needed for state machine logic (state updates persisted on-disk on commit etc.)

Reply:
I'll update this with more clarification